File: Python/program/Vehicle.py
class Vehicle:

    __platNomor = ""
    __merk = ""
    __tahunProduksi = -1
    __warna = ""

    def __init__(self, platNomor="", merk="", tahunProduksi=-1, warna=""):
        self.__platNomor = platNomor
        self.__merk = merk
        self.__tahunProduksi = tahunProduksi
        self.__warna = warna

    def getPlatNomor(self):
        return self.__platNomor

    def setPlatNomor(self, platNomor):
        self.__platNomor = platNomor

    def getMerk(self):
        return self.__merk

    def setMerk(self, merk):
        self.__merk = merk

    def getTahunProduksi(self):
        return self.__tahunProduksi

    def setTahunProduksi(self, tahunProduksi):
        self.__tahunProduksi = tahunProduksi

    def getWarna(self):
        return self.__warna

    def setWarna(self, warna):
        self.__warna = warna
    
    def getByColName(self, colName=""):
        if colName == "platNomor":
            return self.getPlatNomor()
        elif colName == "merk":
            return self.getMerk()
        elif colName == "tahunProduksi":
            return self.getTahunProduksi()
        elif colName == "warna":
            return self.getWarna()

File: CPP/Vehicle.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class Vehicle{
    private:
        string platNomor;
        string merk;
        int tahunProduksi;
        string warna;
    public:

        Vehicle(){

        }

        Vehicle(string platNomor){
            this->platNomor = platNomor;
            this->merk = "";
            this->tahunProduksi = -1;
            this->warna = "";
        }
        Vehicle(string platNomor, string merk){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = -1;
            this->warna = "";
        }
        Vehicle(string platNomor, string merk, int tahunProduksi){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            this->warna = "";
        }
        Vehicle(string platNomor, string merk, int tahunProduksi, string warna){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            this->warna = warna;
        }

        void setPlatNomor(string platNomor) { this->platNomor = platNomor; }
        void setMerk(string merk) { this->merk = merk; }
        void setWarna(string warna) { this->warna = warna; }
        void setTahunProduksi(int tahunProduksi) { this->tahunProduksi = tahunProduksi; }
        
        string getPlatNomor() { return platNomor; }
        string getMerk() { return merk; }
        string getWarna() { return warna; }
        int getTahunProduksi() { return tahunProduksi; }

        string getByColName(string colName){
            if(colName == "platNomor"){ return this->getPlatNomor(); }
            else if(colName == "merk"){ return this->getMerk(); }
            else if(colName == "warna"){ return this->getWarna(); }
            else if(colName == "tahunProduksi"){ return to_string(this->getTahunProduksi()); }
        }

        ~Vehicle(){

        }
};